api/src/main/java/net/md_5/bungee/api/boss/BossBarBuilder.java

```diff
@@ -0,0 +1,186 @@
+     */
+    public BossBarBuilder(BossBarBuilder builder)
+    {
+        this.title = builder.title;
+        this.color = builder.color;
+        this.style = builder.style;
+        this.progress = builder.progress;
+        this.players = builder.players;
+        this.flags = builder.flags;
+    }
+
+    /**
+     * Creates a boss bar builder with the specified title
+     *
+     * @param title the boss bar title you wish to create a boss bar builder with
+     */
+    public BossBarBuilder(BaseComponent[] title)
+    {
+        this();
+        this.title = title;
+    }
+
+    /**
+     * Set the current boss bar's title
+     *
+     * @param title the title you wish to set
+     * @return this BossBarBuilder for chaining
+     */
+    public BossBarBuilder title(BaseComponent[] title)
+    {
+        Preconditions.checkNotNull( title, "title" );
+        this.title = title;
+        return this;
+    }
+
+    /**
+     * Set the current boss bar's color
+     *
+     * @param color the color you wish to set
+     * @return this BossBarBuilder for chaining
+     */
+    public BossBarBuilder color(BarColor color)
+    {
+        this.color = color;
+        return this;
+    }
+
+    /**
+     * Set the current boss bar's style
+     *
+     * @param style the style you wish to set
+     * @return this BossBarBuilder for chaining
+     */
+    public BossBarBuilder style(BarStyle style)
+    {
+        this.style = style;
+        return this;
+    }
+
+    /**
+     * Set the current boss bar's progress. The number specified should be
+     * between 0 and 1 including.
+     *
+     * @param progress the progress you wish to set.
+     * @return this BossBarBuilder for chaining
+     */
+    public BossBarBuilder progress(float progress)
+    {
+        this.progress = progress;
+        return this;
+    }
+
+    /**
+     * Adds a player to the boss bar.
+     *
+     * @param player the player you wish to add
+     * @return this BossBarBuilder for chaining
+     */
+    public BossBarBuilder player(ProxiedPlayer player)
+    {
+        Preconditions.checkNotNull( player, "player" );
+        players.add( player );
+        return this;
+    }
+
+    /**
+     * Adds the specified players to the boss bar
+     *
+     * @param players the players you wish to add
+     * @return this BossBarBuilder for chaining
+     */
+    public BossBarBuilder players(ProxiedPlayer... players)
+    {
+        Preconditions.checkNotNull( players, "players" );
+        this.players.addAll( Arrays.asList( players ) );
+        return this;
+    }
+
+    /**
+     * Adds the specified players to the boss bar
+     *
+     * @param players the players you wish to add
+     * @return this BossBarBuilder for chaining
+     */
+    public BossBarBuilder players(Iterable<ProxiedPlayer> players)
+    {
+        for ( ProxiedPlayer player : players )
+        {
+            player( player );
+        }
+        return this;
+    }
```

Comment:
Inconsistent code. First method converts to `List` and uses `addAll`, second method, where `addAll` would make more sense, its iterating through and calling `player` method to add individually.
I suggest changing the second method to use `addAll` too.

Reply:
addAll accepts a Collection and not a Iterable.

Reply:
You can use Collections.addAll()

Reply:
oh

Reply:
> You can use Collections.addAll()

![cannot be applied](https://user-images.githubusercontent.com/37958105/66337003-f04be900-e946-11e9-939b-7d36adcc8150.png)

Reply:
I've found a way using guava's Iterables class.